### PR TITLE
[huggingface_pytorch][NEURON][build] Huggingface Neuron inference DLC Framework update

### PR DESCRIPTION
**Pull request by philschmid**
*GitHub Issue #, if available:*

### Description

Updating the HF Inference DLC to Neuron 1.10

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [x] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: huggingface/pytorch/buildspec-neuron.yml [hunk 1]
```diff
@@ -2,8 +2,8 @@ account_id: &ACCOUNT_ID <set-$ACCOUNT_ID-in-environment>
 region: &REGION <set-$REGION-in-environment>
 base_framework: &BASE_FRAMEWORK pytorch
 framework: &FRAMEWORK !join [ "huggingface_", *BASE_FRAMEWORK]
-version: &VERSION 1.9.1
-short_version: &SHORT_VERSION 1.9
+version: &VERSION 1.10.1
+short_version: &SHORT_VERSION 1.10
```
Comment (aws-vrnatham) [inline huggingface/pytorch/buildspec-neuron.yml:6]: Please change this to "1.10" and try it. The build should go through fine.

Comment (aws-vrnatham) [inline huggingface/pytorch/buildspec-neuron.yml:6]: Please change this to "1.10" and try it. The build should go through fine.

Comment (philschmid) [reply huggingface/pytorch/buildspec-neuron.yml]: changed it build still failed

Comment (philschmid) [reply huggingface/pytorch/buildspec-neuron.yml]: changed it build still failed

**Comment by philschmid** [comment]
@arjkesh is there a way to get access to the `dlc-pr-huggingface-pytorch-neuron` logs?

File: huggingface/pytorch/inference/docker/1.10/py3/sdk1.18.0/Dockerfile.neuron [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+# Specify accept-bind-to-port LABEL for inference pipelines to use SAGEMAKER_BIND_TO_PORT
+# https://docs.aws.amazon.com/sagemaker/latest/dg/inference-pipeline-real-time.html
+LABEL com.amazonaws.sagemaker.capabilities.accept-bind-to-port=true
+
+ARG MMS_VERSION=1.1.8
+ARG PYTHON=python3.7
+ARG PYTHON_VERSION=3.7.10
+# Pin torch neuron version to sdk1.17.1
+ARG TORCH_NEURON_VERSION=">=1.10.1,<1.10.2"
```
Comment (aws-vrnatham) [inline huggingface/pytorch/inference/docker/1.10/py3/sdk1.18.0/Dockerfile.neuron:14]: Please change this to torch-neuron>=1.10.2,<1.10.3. For future please check the [following](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/release-notes/releasecontent.html)  to find the right framework versions with the neuron sdk release.

**Comment by aws-vrnatham** [comment]
> 



> 

Also please check https://github.com/aws/deep-learning-containers#building-your-image to see if you can build your image locally. This way you don't have to rely on PR for checking build failures (until you get access to the PR logs)

**Comment by arjkesh** [comment]
@philschmid do you have access to the tests?

File: huggingface/pytorch/inference/docker/1.10/py3/sdk1.19.1/Dockerfile.neuron [hunk 3]
```diff
@@ -0,0 +1,139 @@
+FROM ubuntu:18.04
+
+LABEL maintainer="Amazon AI"
+LABEL dlc_major_version="1"
+
+# Specify accept-bind-to-port LABEL for inference pipelines to use SAGEMAKER_BIND_TO_PORT
+# https://docs.aws.amazon.com/sagemaker/latest/dg/inference-pipeline-real-time.html
+LABEL com.amazonaws.sagemaker.capabilities.accept-bind-to-port=true
+
+ARG MMS_VERSION=1.1.8
```
Comment (aws-vrnatham) [inline huggingface/pytorch/inference/docker/1.10/py3/sdk1.19.1/Dockerfile.neuron:10]: Any specific reason for using MMS and not torchserve. Rest of the DLC's are using torchserve

File: huggingface/pytorch/inference/docker/1.10/py3/sdk1.19.1/Dockerfile.neuron [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+ARG PYTHON=python3.7
+ARG PYTHON_VERSION=3.7.10
+# Pin torch neuron version to sdk1.17.1
```
Comment (aws-vrnatham) [inline huggingface/pytorch/inference/docker/1.10/py3/sdk1.19.1/Dockerfile.neuron:13]: Change the comment to sdk1.19.1

File: huggingface/pytorch/buildspec-neuron.yml [hunk 5]
```diff
@@ -2,8 +2,8 @@ account_id: &ACCOUNT_ID <set-$ACCOUNT_ID-in-environment>
 region: &REGION <set-$REGION-in-environment>
 base_framework: &BASE_FRAMEWORK pytorch
 framework: &FRAMEWORK !join [ "huggingface_", *BASE_FRAMEWORK]
-version: &VERSION 1.9.1
-short_version: &SHORT_VERSION 1.9
+version: &VERSION 1.10.1
```
Comment (aws-vrnatham) [inline huggingface/pytorch/buildspec-neuron.yml:5]: Can you please check the version. It has to be 1.10.2. For neuron sdk 1.19.1 only torch-neuron-1.10.2 is released. Please check the following [link](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/release-notes/releasecontent.html#neuron-release-content)

File: huggingface/pytorch/buildspec-neuron.yml [hunk 6]
```diff
@@ -34,9 +34,9 @@ images:
...
     python_version: &DOCKER_PYTHON_VERSION py3
     tag_python_version: &TAG_PYTHON_VERSION py37
-    neuron_sdk_version: &NEURON_SDK_VERSION sdk1.17.1
+    neuron_sdk_version: &NEURON_SDK_VERSION sdk1.19.1
     os_version: &OS_VERSION ubuntu18.04
-    transformers_version: &TRANSFORMERS_VERSION 4.12.3
+    transformers_version: &TRANSFORMERS_VERSION 4.18.0
     tag: !join [ *VERSION, '-', 'transformers', *TRANSFORMERS_VERSION, '-', *DEVICE_TYPE, '-', *TAG_PYTHON_VERSION,"-", *NEURON_SDK_VERSION, '-', *OS_VERSION ]
     docker_file: !join [ docker/, *SHORT_VERSION, /, *DOCKER_PYTHON_VERSION, /, *NEURON_SDK_VERSION, /Dockerfile., *DEVICE_TYPE ]
     context:
```
Comment (aws-vrnatham) [inline huggingface/pytorch/buildspec-neuron.yml:42]: Also please add a buildspec-1-10-2-neuron.yml with the same contents in buildspec-neuron.yml

Comment (philschmid) [reply huggingface/pytorch/buildspec-neuron.yml:42]: why? isn't that something we do when we release 1.11?

File: test/test_utils/__init__.py [hunk 7]
```diff
@@ -1270,6 +1270,25 @@ def get_framework_and_version_from_tag(image_uri):
             "1.8.0": "1.8.0.2.2.2.0",
         },
     },
+    "1.19.1": {
```
Comment (arjkesh) [inline test/test_utils/__init__.py:1273]: @aws-vrnatham suggested adding this to resolve the sanity test failure, which seems to exist as a result of 1.19.1 not being referenced here

**Comment by aws-vrnatham** [comment]
Looks good to me.

**Pull request merged**